File: Nivel_2/Retos/Factorial/FactorialIterativo.java
public class FactorialIterativo {
    public static void main(String[] args) {
        int n = 5;
        System.out.println(String.format("El factorial de %d es %d", n, Factorial(n)));
    }

    public static long Factorial(int n) {
        if (n < 0) {
            return -1;
        }
        long resultado = 1;
        for (int i = 2; i <= n; i++) {
            resultado *= i;
        }
        return resultado;
    }
}

File: Nivel_2/Retos/Factorial/FactorialIterativo.py
def factorial(n):
    if n < 0:
        return -1
    resultado = 1
    for i in range(2, n + 1):
        resultado *= i
    return resultado

def main():
    n = 5
    resultado = factorial(n)
    if resultado == -1:
        print("El número debe ser no negativo.")
    else:
        print(f"El factorial de {n} es {resultado}")

if __name__ == "__main__":
    main()
